### PR TITLE
EVG-13169 Let users move between Issues and Suspect Issues on the UI

### PR DESCRIPTION
**Pull request by malikchaya2**
This change builds on EVG-13170, so I branched off of that branch. When reviewing this PR, please filter commits to only view the ones after the first one. (I will hold off merging, and only merge the new changes after the  EVG-13170 has been merged). 

move_ann

### CODE REVIEW

File: src/components/BuildBaronAndAnnotations/AnnotationTickets.tsx [hunk 1]
```diff
@@ -44,14 +45,23 @@ export const AnnotationTickets: React.FC<Props> = ({
...
       </TitleAndButtons>
-      <AnnotationTicketsTable jiraIssues={tickets} />{" "}
+      <AnnotationTicketsTable
+        jiraIssues={tickets}
+        annotationId={annotationId}
+        taskId={taskId}
+        execution={execution}
+        isIssue={isIssue}
+        userCanModify={userCanModify}
+      />{" "}
```
Comment (SupaJoon) [inline src/components/BuildBaronAndAnnotations/AnnotationTickets.tsx:64]: I don't think this space is required

File: src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx [hunk 2]
```diff
@@ -1,42 +1,205 @@
...
+    const apiIssue = {
+      url,
+      issueKey,
+    };
+    moveAnnotation({ variables: { annotationId, apiIssue, isIssue } });
+  };
+
+  return (
+    <TableWrapper>
+      <ErrorBoundary>
```
Comment (SupaJoon) [inline src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx:175]: is there a specific error you are meaning to catch with the error boundary?

Comment (malikchaya2) [reply src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx:175]: I believe this is part of the first commit, which is not really part of this pr. I branched off of the delete button branch. I explained more in the summary of the PR. Apologies for the confusion!

**Review by SupaJoon — APPROVED** [review]

File: src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx [hunk 3]
```diff
@@ -9,29 +9,36 @@ import { ErrorBoundary } from "components/ErrorBoundary";
...
+  annotationId,
   taskId,
   execution,
   userCanModify,
   jiraIssues,
   isIssue,
 }) => {
   const dispatchBanner = useBannerDispatchContext();
+  const issueString = isIssue ? "issue" : "suspected issue";
+  const icon = isIssue ? <Icon glyph="ArrowDown" /> : <Icon glyph="ArrowUp" />;
```
Comment (khelif96) [inline src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx:41]: `const icon = <Icon glyph={isIssue ? "ArrowDown" : "ArrowUp" />`

File: src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx [hunk 4]
```diff
@@ -61,11 +68,34 @@ export const AnnotationTicketsTable: React.FC<Props> = ({
...
+                isIssue ? "suspected issues" : "issues"
+              }?`}
+              onConfirm={() => {
+                onClickMove(url, issueKey);
+              }}
+              okText="Yes"
+              cancelText="Cancel"
+            >
+              <StyledButton
+                size="small"
```
Comment (khelif96) [inline src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx:84]: This is an enum exported by button you can use `Size.small`

File: src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx [hunk 5]
```diff
@@ -89,11 +119,36 @@ export const AnnotationTicketsTable: React.FC<Props> = ({
     RemoveAnnotationIssueMutationVariables
   >(REMOVE_ANNOTATION, {
     onCompleted: () => {
-      dispatchBanner.successBanner(`Successfully removed issue`);
+      dispatchBanner.successBanner(
+        `Successfully removed ${isIssue ? "issue" : "suspected issue"}`
```
Comment (khelif96) [inline src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx:123]: `issueString` can be used here.

File: src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx [hunk 6]
```diff
@@ -89,11 +119,36 @@ export const AnnotationTicketsTable: React.FC<Props> = ({
...
+      );
     },
     onError(error) {
       dispatchBanner.errorBanner(
-        `There was an error removing the issue: ${error.message}`
+        `There was an error removing the ${
+          isIssue ? "issue" : "suspected issue"
```
Comment (khelif96) [inline src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx:129]: `issueString` can be used here.

File: src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx [hunk 7]
```diff
@@ -89,11 +119,36 @@ export const AnnotationTicketsTable: React.FC<Props> = ({
...
+      dispatchBanner.successBanner(
+        `Successfully moved ${isIssue ? "issue" : "suspected issue"} to ${
+          isIssue ? "suspected issues" : "issues"
+        }`
+      );
+    },
+    onError(error) {
+      dispatchBanner.errorBanner(
+        `There was an error moving the ${
+          isIssue ? "issue" : "suspected issue"
```
Comment (khelif96) [inline src/components/BuildBaronAndAnnotations/AnnotationTicketsTable.tsx:150]: same comment as above.

File: src/components/BuildBaronAndAnnotations/BuildBaron.tsx [hunk 8]
```diff
@@ -34,6 +34,7 @@ const BuildBaron: React.FC<Props> = ({
     {bbData && (
       <BuildBaronContent
         bbData={bbData.buildBaron}
+        annotationId={annotation?.id}
```
Comment (khelif96) [inline src/components/BuildBaronAndAnnotations/BuildBaron.tsx:37]: Is `annotation` nullable? If so what happens if its undefined?
The props for BuildBaronContent apply that `annotationId` is required

**Comment by khelif96** [comment]
Try merging in master for the failing tests

File: cypress/integration/task/annotations.ts [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+const taskRoute = `/task/${taskWithAnnotations}/annotations`;
+
+describe("Task Annotation Tab", () => {
+  before(() => {
+    cy.login();
+  });
+
+  beforeEach(() => {
+    cy.preserveCookies();
+    cy.listenGQL();
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply cypress/integration/task/annotations.ts:14]: I believe this is part of EVG-13170, not this ticket.

File: src/components/BuildBaronAndAnnotations/BuildBaron.tsx (hunk 8 above)
Comment (malikchaya2) [reply src/components/BuildBaronAndAnnotations/BuildBaron.tsx:37]: It is used by the moveAnnotation mutation, and that button only shows up in the table row of issues or suspected issues. So it wouldn't surface unless there is an issue or a suspected issue.

**Pull request merged**